src/shared/wizards/wizardForm.ts

```diff
@@ -97,25 +169,25 @@ export class WizardForm<TState extends Partial<Record<keyof TState, unknown>>> {
         this.body = this.createWizardForm()
     }
 
+    // note: this solves the dependency tree, so even though it's an accessor it's a pretty expensive one
     public get properties(): string[] {
-        return [...this.formData.keys()].sort(this.compareOrder.bind(this))
+        return this.resolveDependencies()
     }
 
-    public getPrompterProvider(prop: string): PrompterProvider<TState, any> | undefined {
+    public getPrompterProvider(prop: string): PrompterProvider<TState, any, any> | undefined {
```

Comment:
I feel like this is an awful lot of `any`s for a public function. Can we tie this to the class' types?